### PR TITLE
Add opnode makefile, lints, and go1.18

### LINKED ISSUE #205: Consolidate Makefiles
**Description:** Decide if we want a makefile at the root of the repo or in the root of each project. See discussion in #197

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**
This does the following:
- Add a makefile to the opnode package
- Add the `misspell` lint to the makefiles, docs, and CI
- Use the supplied git versions in the opnode
- Quiet logging during opnode tests
- Require go 1.18

**Notes**

The misspell lint is good, but not perfect. The only false positive that I found was it was trying to correct `strat` to `start`.

**Metadata**
- Fixes #205

### CODE REVIEW

**Review by protolambda — CHANGES_REQUESTED** [review]
Very nice, looking forward to start using Go fuzzing. Don't forget to update the CI Go version though.

File: .github/workflows/go-ci.yml [hunk 1]
```diff
@@ -21,10 +21,10 @@ jobs:
       - name: golangci-lint
```
[the remark this answers is not in the document]
Comment (trianglesphere) [reply .github/workflows/go-ci.yml:21]: yes, important catch

[the remark this answers is not in the document]
Comment (trianglesphere) [reply .github/workflows/go-ci.yml:21]: fixed.

**Review by protolambda — APPROVED** [review]
LGTM

**Pull request merged**